app/models/patient/consent_status.rb

```diff
@@ -24,8 +24,66 @@ class Patient::ConsentStatus < ApplicationRecord
   belongs_to :patient
   belongs_to :programme
 
+  has_many :consents,
+           -> do
+             not_invalidated.response_provided.eager_load(:parent, :patient)
```

Comment:
Just wondering out loud, but is there any reason for the `consents` association not to always be scoped to `programme`? It seems odd that you could pull up the consents and get unrelated ones for this object. Or is there a technical reason not to?

I also see that `ConsentGrouper` does it's own filtering as part of it's interface, which makes sense, but even so it feels odd to me that the `ConsentStatus` for a particular programme could return unrelated consents.

Reply:
Basically I tested both approaches, and scoping this to the program ends up needing to make more queries and is slower overall.

If we scope it to the programme, Rails ends up pre-loading the consents 3 times (for each programme) making 3 queries to the database for each batch of consents statuses we want to update.

Whereas if instead we handle the programme filtering in memory, only one query to the database is made, and overall the performance is better and memory usage is lower.

Originally I wanted to scope these (and there's a few others in the other status classes) but I found this to the most performant approach in the end.